tests/test_l1dbproto.py

```diff
@@ -0,0 +1,475 @@
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <http://www.lsstcorp.org/LegalNotices/>.
+#
+
+from __future__ import absolute_import, division, print_function
+
+import os
+import numpy as np
+import tempfile
+import unittest
+
+from lsst.ap.association import \
+    AssociationL1DBProtoConfig, \
+    AssociationL1DBProtoTask, \
+    make_minimal_dia_source_schema
+from lsst.afw.cameraGeom.testUtils import DetectorWrapper
+import lsst.afw.image as afwImage
+import lsst.afw.image.utils as afwImageUtils
+import lsst.afw.geom as afwGeom
+import lsst.afw.table as afwTable
+import lsst.daf.base as dafBase
+import lsst.pipe.base as pipeBase
+import lsst.utils.tests
+
+
+def create_test_points(point_locs_deg,
+                       start_id=0,
+                       schema=None,
+                       scatter_arcsec=1.0,
+                       indexer_ids=None,
+                       associated_ids=None):
+    """Create dummy DIASources or DIAObjects for use in our tests.
+
+    Parameters
+    ----------
+    point_locs_deg : array-like (N, 2) of `float`s
+        Positions of the test points to create in RA, DEC.
+    start_id : `int`
+        Unique id of the first object to create. The remaining sources are
+        incremented by one from the first id.
+    schema : `lsst.afw.table.Schema`
+        Schema of the objects to create. Defaults to the DIASource schema.
+    scatter_arcsec : `float`
+        Scatter to add to the position of each DIASource.
+    indexer_ids : `list` of `ints`s
+        Id numbers of pixelization indexer to store. Must be the same length
+        as the first dimension of point_locs_deg.
+    associated_ids : `list` of `ints`s
+        Id numbers of associated DIAObjects to store. Must be the same length
+        as the first dimension of point_locs_deg.
+
+    Returns
+    -------
+    test_points : `lsst.afw.table.SourceCatalog`
+        Catalog of points to test.
+    """
+    if schema is None:
+        schema = make_minimal_dia_source_schema()
+    sources = afwTable.SourceCatalog(schema)
+
+    for src_idx, (ra, dec,) in enumerate(point_locs_deg):
+        src = sources.addNew()
+        src['id'] = src_idx + start_id
+        coord = afwGeom.SpherePoint(ra, dec, afwGeom.degrees)
+        if scatter_arcsec > 0.0:
+            coord = coord.offset(
+                np.random.rand() * 360 * afwGeom.degrees,
+                np.random.rand() * scatter_arcsec * afwGeom.arcseconds)
+        if indexer_ids is not None:
+            src['pixelId'] = indexer_ids[src_idx]
+        if associated_ids is not None:
+            src['diaObjectId'] = associated_ids[src_idx]
+        src.setCoord(coord)
+
+    return sources
+
+
+class TestAssociationDBSqlite(unittest.TestCase):
+
+    def setUp(self):
+        """Initialize an empty database.
+        """
+
```

Comment:
I see you use np.random below. Unit tests must be deterministic, so add `np.random.seed(1)` here, where 1 is your favorite number.